subprojects/core/src/main/java/org/gradle/process/internal/worker/child/BootstrapSecurityManager.java

```diff
@@ -86,7 +86,7 @@ public void checkPermission(Permission permission) {
             SecurityManager securityManager;
             try {
                 Class<?> aClass = systemClassLoader.loadClass(securityManagerType);
-                securityManager = (SecurityManager) aClass.newInstance();
+                securityManager = (SecurityManager) aClass.getConstructor().newInstance();
```

Comment:
Do we need to expect the security manager to have a public constructor here? I am not sure...

Reply:
I thought this `securityManagerType` is controlled by us via `sendUTF`... but after I checked out the code, it seems not... https://github.com/gradle/gradle/blob/aa806d9ac84ac6ca04b73698cff84c10b5990eb9/subprojects/core/src/main/java/org/gradle/process/internal/worker/child/ApplicationClassesInSystemClassLoaderWorkerImplementationFactory.java#L88